### PR TITLE
task complited

### PR DESCRIPTION
**Pull request by david-vozniak**

### CODE REVIEW

File: src/main/java/core/basesyntax/Colors.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package main.java.core.basesyntax;
+
+public enum Colors {
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/Colors.java:3]:
```suggestion
public enum Color {

```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+
+import java.util.Random;
 
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static String getRandomColor() {
+        int index = new Random().nextInt(Colors.values().length);
+        Colors colors = Colors.values()[index];
+        return String.valueOf(colors);
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/ColorSupplier.java:9]:
```suggestion
        return Colors.values()[index].name();
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package main.java.core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public static String getRandomBall() {
+        int ballIndex = new Random().nextInt(100);
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/Lottery.java:7]: make magic number constant

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 4]
```diff
@@ -1,7 +1,11 @@
-package core.basesyntax;
+package main.java.core.basesyntax;
+
+import java.util.Random;
 
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static String getRandomColor() {
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: do not make it static, create an instance of supplier where you need it

File: src/main/java/core/basesyntax/Lottery.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package main.java.core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public static String getRandomBall() {
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/Lottery.java:6]:
```suggestion
    public Ball getRandomBall() {

```

File: src/main/java/core/basesyntax/Application.java [hunk 6]
```diff
@@ -1,7 +1,10 @@
-package core.basesyntax;
+package main.java.core.basesyntax;
 
-public class Application {
+public class Application extends Lottery {
     public static void main(String[] args) {
+        System.out.println(getRandomBall());
+        System.out.println(getRandomBall());
+        System.out.println(getRandomBall());
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/Application.java:7]: use cicle

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Comment by david-vozniak** [comment]
could you check my code, please?

File: src/main/java/core/basesyntax/Application.java [hunk 7]
```diff
@@ -1,7 +1,11 @@
 package core.basesyntax;
 
-public class Application {
+public class Application extends Lottery {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        for (int i = 0; i < 3; i++) {
+            System.out.println(new Lottery().getRandomBall());
+        }
+
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Application.java:9]:
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 8]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static String getRandomColor() {
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
    public String getRandomColor() {
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public Ball getRandomBall() {
+        final int bound = 100;
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Lottery.java:7]: magic number

File: src/main/java/core/basesyntax/Lottery.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public Ball getRandomBall() {
+        final int bound = 100;
+        int ballIndex = new Random().nextInt(bound);
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/Lottery.java:8]: don`t create random object inside method

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Comment by david-vozniak** [comment]
I hope my edits make some sense. Otherwise, I don’t quite understand what is required of me

File: src/main/java/core/basesyntax/Application.java [hunk 11]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
-public class Application {
+public class Application extends Lottery {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        for (int i = 0; i < 3; i++) {
+            System.out.println(new Lottery().getRandomBall());
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Application.java:7]: Don't create Lottery object in a loop

File: src/main/java/core/basesyntax/Application.java [hunk 12]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
-public class Application {
+public class Application extends Lottery {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        for (int i = 0; i < 3; i++) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Application.java:6]: 3 is a magic number, create constant

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 13]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        int index = new Random().nextInt(Color.values().length);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: Random should be class-level variable

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 14]
```diff
@@ -1,7 +1,10 @@
...
+        return Color.values()[index].toString();
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]: don't use toString for enum
```suggestion
        return Color.values()[index].name();
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 15]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public static final int MAX_BOUND = 100;
+
+    public int getRandomNumber() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:8]:
```suggestion
    private int getRandomNumber() {
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 16]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public static final int MAX_BOUND = 100;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:6]:
```suggestion
    public static final int MAX_BOUND = 100;
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();
```

File: src/main/java/core/basesyntax/Lottery.java (hunk 15 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/Lottery.java:8]: Replace after public method

File: src/main/java/core/basesyntax/Lottery.java [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+
+public class Lottery extends ColorSupplier {
+    public static final int MAX_BOUND = 100;
+
+    public int getRandomNumber() {
+        return new Random().nextInt(MAX_BOUND);
+    }
+
+    public Ball getRandomBall() {
+        return new Ball(getRandomColor(), getRandomNumber());
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:13]:
```suggestion
        return new Ball(colorSupplier.getRandomColor(), getRandomNumber());

```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]
fix prev comments

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 18]
```diff
@@ -1,7 +1,11 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
+    private final int randomNumber = new Random().nextInt(Color.values().length);
```
[the remark this answers is not in the document]
Comment (david-vozniak) [reply src/main/java/core/basesyntax/ColorSupplier.java]: 'Random should be class-level variable'. That was the previous mentor request. That`s why I made "Random" a variable.

**Comment by david-vozniak** [comment]
Why do I get review from different mentor, when I made request only to previous. It doesn't appears to be comfortable. Can we stick with only a couple?

File: src/main/java/core/basesyntax/Lottery.java [hunk 19]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public static final int MAX_BOUND = 100;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    private int getRandomNumber() {
```
[the remark this answers is not in the document]
Comment (MentorMateAcademy) [reply src/main/java/core/basesyntax/Lottery.java:10]: not fixed, you have private above public method

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 20]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
+
     public String getRandomColor() {
-        return null;
+        return Color.values()[new Random().nextInt(Color.values().length)].name();
```
Comment (MentorMateAcademy) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]:
```suggestion
        return Color.values()[random.nextInt(Color.values().length)].name();
```

Comment (MentorMateAcademy) [reply src/main/java/core/basesyntax/ColorSupplier.java:8]: Make it like this 
image

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request closed without merging**